File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/dtos/menu/MenuHallProductResponseDTO.java
package com.robinfood.core.dtos.menu;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@AllArgsConstructor
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class MenuHallProductResponseDTO {

    private final Long articleId;
    private final Long brandId;
    private final String description;
    private final BigDecimal discount;
    private final Long displayType;
    private final List<MenuProductGroupDTO> groups;
    private final Long id;
    private final String image;
    private final String name;
    private final Long parentId;
    private final Integer position;
    private final BigDecimal price;
    private final String productFlow;
    private final Long sizeId;
    private final String sku;
    private final List<String> tags;
    private final Long type;
    private final String typeName;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/menu/response/MenuHallProductResponseDTO.kt
package com.robinfood.core.dtos.menu.response

import java.math.BigDecimal

data class MenuHallProductResponseDTO(
        val articleId: Long,
        val brandId: Long,
        val description: String,
        val discount: BigDecimal,
        val displayType: Long,
        val groups: List<MenuProductGroupDTO>,
        val id: Long,
        val image: String,
        val name: String,
        val parentId: Long,
        val position: Int,
        val price: BigDecimal,
        val productCategoryId: Long,
        val productFlow: String,
        val sizeId: Long,
        val sku: String,
        val tags: List<String>,
        val type: Long,
        val typeName: String
)
